### PR TITLE
[IMP] open_academy: Wizard to register attendees for sessions T#59013

### PR DESCRIPTION
**Pull request by geraspace**
- Add wizard to register new attendees to one or various sessions

### CODE REVIEW

**Comment by geraspace** [comment]
@edy1192 here is a short PR for the wizard. Thanks!

File: open_academy/wizards/open_academy_wizard.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from odoo import models, fields, api
+
+
+class OpenAcademyWizard(models.TransientModel):
+    _name = "open.academy.wizard"
+    _description = "Quick registration of attendees to sessions"
+
+    def _default_sessions(self):
+        return self.env["open.academy.session"].browse(self._context.get("active_ids"))
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard.py:14]: Move this method after field's declaration. See [guidelines](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#37models) `In a Model attribute order should be` 

When you move this method you'll get an error. You just use the lambda function to call the method in field's attribute.

`default=lambda self: self._default_sessions()`

File: open_academy/wizards/open_academy_wizard.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    session_wizard_ids = fields.Many2many('open.academy.session', default=_default_sessions)
+    attendee_wizard_ids = fields.Many2many("res.partner")
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard.py]: Don't add `wizard` to field's name.

```suggestion
    session_ids = fields.Many2many('open.academy.session', default=_default_sessions)
    attendee_ids = fields.Many2many("res.partner")
```

File: open_academy/wizards/open_academy_wizard_views.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    <record id="open_academy_wizard_view_form" model="ir.ui.view">
+        <field name="name">open.academy.wizard.view.form</field>
+        <field name="model">open.academy.wizard</field>
+        <field name="arch" type="xml">
+            <form>
+                <field name="session_wizard_ids" string="Session"/>
+                <field name="attendee_wizard_ids" string="Attendees"/>
+                <footer>
+                    <button name="subscribe" type="object" string="Finish" class="oe_highlight"/>
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard_views.xml]:
```suggestion
                <footer>
                    <button name="subscribe" type="object" string="Add Attendees" class="oe_highlight"/>
```

File: open_academy/wizards/open_academy_wizard_views.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                        or
+                    <button special="cancel" string="Cancel"/>
+                </footer>
+            </form>
+        </field>
+    </record>
+
+    <record id="launch_session_wizard" model="ir.actions.act_window">
+        <field name="name">Add atendees to session</field>
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard_views.xml]:
```suggestion
        <field name="name">Add attendees to sessions</field>
```

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by geraspace** [comment]
Hi, @edy1192 I think it's ready!

File: open_academy/i18n/es_MX.po [hunk 5]
```diff
@@ -0,0 +1,193 @@
+# Translation of Odoo Server.
```
Comment (edy1192) [inline open_academy/i18n/es_MX.po]: Use es.po file please. Remember what I said yesterday.

File: open_academy/wizards/open_academy_wizard.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+from odoo import models, fields
+
+
+class OpenAcademyWizard(models.TransientModel):
+    _name = "open.academy.wizard"
+    _description = "Quick registration of attendees to sessions"
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard.py]:
```suggestion
    _description = "open.academy.wizard"
```

File: open_academy/wizards/open_academy_wizard.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    session_ids = fields.Many2many('open.academy.session',
+                                   default=lambda self: self._default_sessions())
+    attendee_ids = fields.Many2many("res.partner")
+
+    def _default_sessions(self):
+        return self.env["open.academy.session"].browse(self._context.get("active_ids"))
+
+    def subscribe(self):
+        for session in self.session_wizard_ids:
+            session.attendee_ids |= self.attendee_wizard_ids
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard.py]: Did you test this?

You changed the field's name, so this is wrong.

File: open_academy/wizards/open_academy_wizard_views.xml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="open_academy_wizard_view_form" model="ir.ui.view">
+        <field name="name">open.academy.wizard.view.form</field>
+        <field name="model">open.academy.wizard</field>
+        <field name="arch" type="xml">
+            <form>
+                <field name="session_wizard_ids" string="Session"/>
+                <field name="attendee_wizard_ids" string="Attendees"/>
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard_views.xml]: Same here. These fields do not exists anymore.

Please test your changes before ask for revision.

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by geraspace** [comment]
You are right, I don't know why I did not test, @edy1192. Now it's working, and I also added the translation to the commit message. Thanks!

File: open_academy/i18n/es.po [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: open_academy
+#: code:addons/open_academy/models/open_academy_session.py:0
+#, python-format
+msgid "A session's instructor can't be an attendee."
+msgstr "Un instructor no puede ser un asistente"
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "Un instructor no puede ser un asistente."
```

File: open_academy/i18n/es.po [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_course__create_uid
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_session__create_uid
+msgid "Created by"
+msgstr "Creado por"
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_course__create_date
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_session__create_date
+msgid "Created on"
+msgstr "Fecha de creación"
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "Creado el"
```

File: open_academy/i18n/es.po [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+#. module: open_academy
+#: model:ir.model.fields,help:open_academy.field_open_academy_session__duration
+msgid "Duration in days"
+msgstr "Duración en días"
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_course__id
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_session__id
+msgid "ID"
+msgstr "ID"
```
Comment (edy1192) [inline open_academy/i18n/es.po]: When terms are similar between languages the msgtr is empty.

```suggestion
msgstr ""
```

File: open_academy/i18n/es.po [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+#, python-format
+msgid "Incorrect values"
+msgstr "Valores incorrectos"
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_session__instructor_id
+#: model:ir.model.fields,field_description:open_academy.field_res_partner__instructor
+#: model:ir.model.fields,field_description:open_academy.field_res_users__instructor
+msgid "Instructor"
+msgstr "Instructor"
```
Comment (edy1192) [inline open_academy/i18n/es.po]: Same here, leave it empty please.

File: open_academy/i18n/es.po [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_course____last_update
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_session____last_update
+msgid "Last Modified on"
+msgstr "Fecha de última modificación"
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "Última modificación el"
```

File: open_academy/i18n/es.po [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_course__write_uid
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_session__write_uid
+msgid "Last Updated by"
+msgstr "Última actualización por"
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_course__write_date
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_session__write_date
+msgid "Last Updated on"
+msgstr "Fecha de última actualización"
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "Última actualización el"
```

File: open_academy/i18n/es.po [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+
+#. module: open_academy
+#: model:res.groups,name:open_academy.open_academy_group_manager
+msgid "Manager"
+msgstr "Gestor"
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "Administrador"
```

File: open_academy/i18n/es.po [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+
+#. module: open_academy
+#: model:ir.model.fields,field_description:open_academy.field_open_academy_session__taken_seats
+msgid "Taken Seats"
+msgstr "Asientos ocupados"
+
+#. module: open_academy
+#: model:ir.model.constraint,message:open_academy.constraint_open_academy_course_name_description
+msgid "The course title must be different to the description."
+msgstr "El título del curso debe ser diferente a la descripción"
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "El título del curso debe ser diferente a la descripción."
```

File: open_academy/i18n/es.po [hunk 17]
```diff
@@ -0,0 +1,193 @@
...
+
+#. module: open_academy
+#: model:ir.model.constraint,message:open_academy.constraint_open_academy_course_name_unique
+msgid "The course title must be unique."
+msgstr "El título del curso debe ser único"
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "El título del curso debe ser único."
```

File: open_academy/i18n/es.po [hunk 18]
```diff
@@ -0,0 +1,193 @@
...
+
+#. module: open_academy
+#: code:addons/open_academy/models/open_academy_session.py:0
+#, python-format
+msgid "The number of attendees can not be greater than the seats available."
+msgstr "El numero de asistentes no puede ser mayor al número de asientos disponibles."
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "El número de asistentes no puede ser mayor al número de asientos disponibles."
```

File: open_academy/i18n/es.po [hunk 19]
```diff
@@ -0,0 +1,193 @@
...
+
+#. module: open_academy
+#: code:addons/open_academy/models/open_academy_session.py:0
+#, python-format
+msgid "The number of seats can not be less than 0."
+msgstr "El número de asientos no puede ser negativo."
```
Comment (edy1192) [inline open_academy/i18n/es.po]:
```suggestion
msgstr "El número de asientos no puede ser menor a 0."
```

File: open_academy/wizards/open_academy_wizard.py (hunk 6 above)
Comment (edy1192) [reply open_academy/wizards/open_academy_wizard.py]: This change is missing

File: open_academy/wizards/open_academy_wizard.py [hunk 20]
```diff
@@ -0,0 +1,18 @@
+from odoo import models, fields
+
+
+class OpenAcademyWizard(models.TransientModel):
+    _name = "open.academy.wizard"
+    _description = "Quick registration of attendees to sessions"
+
+    session_ids = fields.Many2many('open.academy.session',
+                                   default=lambda self: self._default_sessions())
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard.py]: See this https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#33idioms

```suggestion
    session_ids = fields.Many2many(
                                   'open.academy.session',
                                   default=lambda self: self._default_sessions())
```

File: open_academy/wizards/open_academy_wizard_views.xml [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+
+    <record id="open_academy_wizard_view_form" model="ir.ui.view">
+        <field name="name">open.academy.wizard.view.form</field>
+        <field name="model">open.academy.wizard</field>
+        <field name="arch" type="xml">
+            <form>
+                <field name="session_ids" string="Session"/>
+                <field name="attendee_ids" string="Attendees"/>
+                <footer>
+                    <button name="subscribe" type="object" string="Add attendees" class="oe_highlight"/>
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard_views.xml:12]: This term is missing in translation.

File: open_academy/wizards/open_academy_wizard_views.xml [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+                        or
+                    <button special="cancel" string="Cancel"/>
+                </footer>
+            </form>
+        </field>
+    </record>
+
+    <record id="launch_session_wizard" model="ir.actions.act_window">
+        <field name="name">Add attendees to sessions</field>
```
Comment (edy1192) [inline open_academy/wizards/open_academy_wizard_views.xml:21]: Same here, missing in es.po file

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by geraspace** [comment]
I hope it's okay now, @edy1192. Thanks :)

File: open_academy/i18n/es.po [hunk 23]
```diff
@@ -0,0 +1,252 @@
...
+"                                    Fecha de inicio:"
+
+#. module: open_academy
+#: model_terms:ir.ui.view,arch_db:open_academy.open_academy_session_view_kanban
+msgid ""
+"<br/>\n"
+"                                    duration:"
+msgstr ""
+"<br/>\n"
+"                                    Duración:"
```
Comment (edy1192) [inline open_academy/i18n/es.po:34]:
```suggestion
"                                    duración:"
```

**Review by edy1192 — CHANGES_REQUESTED** [review]

**Comment by geraspace** [comment]
I think those should have been uppercased, :) thanks, @edy1192. It can be checked now.

**Review by edy1192 — APPROVED** [review]

**Pull request merged**